bazel/ray_deps_setup.bzl

```diff
@@ -1,128 +1,198 @@
 load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository", "new_git_repository")
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 
-def ray_deps_setup():
-    RULES_JVM_EXTERNAL_TAG = "2.10"
-    RULES_JVM_EXTERNAL_SHA = "1bbf2e48d07686707dd85357e9a94da775e1dbd7c464272b3664283c9c716d26"
+def github_repository(**kwargs):
```

Comment:
How about naming it `ray_dependency_repository()`, because we use this method not only declaring github dependencies, but also declaring other type dependencies.

Reply:
I'm not sure what you mean -- what other type dependencies is this used for? Note that this is intended strictly for GitHub, and it's not specific to Ray in any way -- you can't in general switch between ZIP archives and git clones; it's only possible on GitHub because they have a well-defined URL format.